Automattic/sensei
# Sensei on Dotcom - Connect Sensei Home tasks statuses with Calypso Launchpad tasks statuses

Fixes [#Automattic/sensei-pro#1893](https://github.com/Automattic/sensei-pro/issues/1893)

### Changes proposed in this Pull Request

* Adds a `Sensei_Utils::is_atomic_platform` method that tells if the website is hosted on wp.com atomic site. Inspired from [here](https://github.com/Automattic/jetpack/blob/ef06c3b8b53be427947f8f67321a319caf2668f0/projects/packages/status/src/class-host.php#L39-L41).
* Adds a `launchpad_checklist_tasks_statuses` wp option with statuses for Sensei Home tasks.

### Testing instructions


* Complete steps for testing instructions from [wp-calypso pr](https://github.com/Automattic/wp-calypso/pull/70066).
* If you followed the steps from wp-calypso pr instructions, then it means you've published a course by now.
* Now visit the Sensei Launchpad again. We saved the url for the launchpad in previous steps.
* Confirm that even though you've published a course, the "Publish your first Course" is not marked as complete. The changes in this PR should fix this. To apply these changes we need to manually edit the file plugins in your newly created atomic website on wp.com.
* Go to **Tools > Plugin File Editor** on your new atomic website.
* On top right you'll need to select a plugin that you want to edit. It says **Select plugin to edit:**. Then press **Select**.
* In the **Plugin Files** navigation on right side select **includes > class-sensei-utils.php**. Once selected it should say **Editing sensei-lms/includes/class-sensei-utils.php (active)** at the top.
* Replace the contents of this file with [this one](https://github.com/Automattic/sensei/blob/add/connect-sensei-home-tasks-with-calypso-launchpad-tasks/includes/class-sensei-utils.php). Click **Update File**.
* In the **Plugin Files** navigation on right side select **includes > admin > home > tasks > class-sensei-home-tasks-provider.php**. Once selected it should say **Editing sensei-lms/includes/admin/home/tasks/class-sensei-home-tasks-provider.php (active)** at the top.
* Replace the contents of this file with [this one](https://github.com/Automattic/sensei/blob/add/connect-sensei-home-tasks-with-calypso-launchpad-tasks/includes/admin/home/tasks/class-sensei-home-tasks-provider.php). Click **Update File**.
* Edit the course post that you published in earlier steps and click **Update**.
* Visit the Sensei Launchpad screen by visiting the url that we saved earlier and confirm that the "Publish your first Course" task is marked as complete.
